var/spack/repos/builtin/packages/dpcpp/package.py

```diff
@@ -16,80 +17,122 @@ class Dpcpp(CMakePackage):
     git = "https://github.com/intel/llvm.git"
 
     version("develop", branch="sycl")
-    version("2021.09", commit="bd68232bb96386bf7649345c0557ba520e73c02d")
-    version("2021.12", commit="27f59d8906fcc8aece7ff6aa570ccdee52168c2d")
+    version(
+        "2023-03",
+        sha256="ca85303d712c58316a91a7c97f7c78fa563a29f1669d8b2368d0c8bd92a63068",
+        url="https://github.com/intel/llvm/tarball/cb91c232",
+    )
+    version("2021.09", commit="bd68232bb96386bf7649345c0557ba520e73c02d", deprecated=True)
+    version("2021.12", commit="27f59d8906fcc8aece7ff6aa570ccdee52168c2d", deprecated=True)
 
     maintainers("ravil-mobile")
+
     variant("cuda", default=False, description="switch from OpenCL to CUDA")
-    variant("rocm", default=False, description="switch from OpenCL to ROCm")
+    variant("hip", default=False, description="switch from OpenCL to HIP")
     variant(
-        "rocm-platform",
+        "hip-platform",
         default="AMD",
         values=("AMD", "NVIDIA"),
         multi=False,
-        description="choose ROCm backend",
+        description="choose HIP backend",
     )
-    variant("openmp", default=False, description="build with OpenMP without target offloading")
-    variant("esimd-cpu", default=False, description="build with ESIMD_CPU support")
+    variant("esimd-emulator", default=False, description="build with ESIMD emulation support")
     variant("assertions", default=False, description="build with assertions")
     variant("docs", default=False, description="build Doxygen documentation")
     variant("werror", default=False, description="treat warnings as errors")
-    variant("shared", default=False, description="build shared libraries")
-    variant("remangle_libclc", default=True, description="remangle libclc gen. variants")
+    variant("shared-libs", default=False, description="build shared libraries")
     variant("lld", default=False, description="use LLD linker for build")
+    variant("fusion", default=True, description="Enable the kernel fusion JIT compiler")
+    variant(
+        "security_flags",
+        default="none",
+        values=("none", "default", "sanitize"),
+        multi=False,
+        description="Enables security flags for compile & link",
+    )
+    variant(
+        "llvm-external-projects",
+        values=str,
+        default="none",
+        description="Add external projects to build. Add as a comma seperated list.",
+    )
 
     depends_on("cmake@3.16.2:", type="build")
     depends_on("ninja@1.10.0:", type="build")
 
-    depends_on("cuda@10.2.0:11.4.999", when="+cuda")
+    depends_on("cuda@10.2.0:", when="+cuda")
```

Comment:
Can we reuse anything from the `CudaPackage` and `RocmPackage` base classes?

Reply:
Potentially could be useful use the `CudaPackage` and `RocmPackage`. Although I don't think there is any use of the `cuda_arch` and `amdgpu_target` variants as we are simply just building the backend. I think `RocmPackage` also enforces `amdgpu_target` by default. 

Also the `intel/llvm` documentation recommends to use at least CUDA 11.6 as their were issues some math built-ins when using the optimization flags. Is there any way we can recommend this to the concertizer? Or at least show up when doing a `spack info`.